agent/yubiagent/slot.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 Yahoo Inc.
+// Licensed under the terms of the Apache License 2.0. Please see LICENSE file in project root for terms.
+
+package yubiagent
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	"github.com/theparanoids/crypki/proto"
+	"github.com/theparanoids/ysshra/attestation/yubiattest"
+	"github.com/theparanoids/ysshra/keyid"
+	keyutil "github.com/theparanoids/ysshra/sshutils/key"
+	"golang.org/x/crypto/ssh"
+)
+
+// SlotAgent wraps the yubiagent and stores the information of a key slot in the agent.
+type SlotAgent struct {
+	yubiAgent YubiAgent
+	code      string                                // the code of the key slot, e.g. "9a", "9e"
+	public    ssh.PublicKey                         // the public key stored in the key slot
+	attest    *x509.Certificate                     // the attestation certificate of the key slot
+	policy    keyid.TouchPolicy                     // the touch policy of the key slot
+	csrs      []*proto.SSHCertificateSigningRequest // the certificates signing requests that are backed by the key slot
+}
+
+// NewSlotAgent returns the key slot agent.
+func NewSlotAgent(yubiAgent YubiAgent, code string) (*SlotAgent, error) {
+	attestationCert, err := yubiAgent.AttestSlot(code)
+	if err != nil {
+		return nil, err
+	}
+	// Due to Infineon Technologies' RSA key generation issue, we do not support RSA
+	// certificates generated by YubiKeys with firmware between 4.2.6-4.3.4.
+	// Ref: https://www.yubico.com/support/security-advisories/ysa-2017-01/
+	if attestationCert.PublicKeyAlgorithm == x509.RSA && isAffectedVersion(attestationCert.Extensions) {
```

Comment:
Please also put the CVE number in the comment. And for the `isAffectedVersion`, maybe rename to `isCVE201715361Affected`.